examples/karaoke.rs

```diff
@@ -18,29 +23,57 @@ const SAMPLE_RATE: f64 = 48_000.0;
 // webrtc-audio-processing expects a 10ms chunk for each process call.
 const FRAMES_PER_BUFFER: u32 = 480;
 
-fn create_processor(
-    num_capture_channels: i32,
-    num_render_channels: i32,
-) -> Result<Processor, Error> {
-    let mut processor = Processor::new(&InitializationConfig {
-        num_capture_channels,
-        num_render_channels,
-        ..InitializationConfig::default()
-    })?;
+#[derive(Debug, StructOpt)]
+struct Args {
+    #[structopt(short, long, default_value = "examples/aec-configs/config.json5")]
+    config_file: PathBuf,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(default)]
+struct AppConfig {
+    num_capture_channels: usize,
+    num_render_channels: usize,
+    #[serde(default)]
+    config: Config,
+    #[serde(default)]
+    aec3: EchoCanceller3Config,
+}
+
+impl Default for AppConfig {
+    fn default() -> Self {
+        Self {
+            num_capture_channels: 1,
+            num_render_channels: 1,
+            config: Config::default(),
+            aec3: EchoCanceller3Config::default(),
+        }
+    }
+}
 
-    // High pass filter is a prerequisite to running echo cancellation.
-    let config = Config {
-        echo_cancellation: Some(EchoCancellation {
-            suppression_level: EchoCancellationSuppressionLevel::Low,
-            stream_delay_ms: Some(0),
-            enable_delay_agnostic: true,
-            enable_extended_filter: true,
-        }),
-        enable_high_pass_filter: true,
-        ..Config::default()
-    };
-    processor.set_config(config);
+impl AppConfig {
+    fn from_file_with_defaults(path: &PathBuf) -> Result<Self, Error> {
```

Comment:
Not sure if this is a more accurate name or not, but the code kind of reads like it.

```suggestion
    fn from_file_or_default(path: &PathBuf) -> Result<Self, Error> {
```